tool/src/js/conversions.rs

```diff
@@ -305,7 +275,26 @@ impl fmt::Display for InvocationIntoJs<'_> {
                             let diplomat_receive_buffer: ast::Ident = "diplomat_receive_buffer".into();
                             writeln!(f, "const {diplomat_receive_buffer} = wasm.diplomat_alloc({size}, {align});")?;
                             writeln!(f, "{};", self.invocation.complex(&diplomat_receive_buffer))?;
-                            writeln!(f, "const out = new {}({diplomat_receive_buffer});", strct.name)?;
+                            writeln!(
+                                f,
+                                "const out = new {}({});",
+                                strct.name,
+                                display::expr(|f| {
+                                    diplomat_receive_buffer.fmt(f)?;
+                                    for inputs in strct.lifetimes.names().map(|name| &self.lifetimes[name][..]) {
+                                        write!(f, ", [{}]", display::expr(|f| {
+                                            if let Some((first, rest)) = inputs.split_first() {
+                                                first.fmt(f)?;
+                                                for input in rest {
+                                                    write!(f, ", {input}")?;
```

Comment:
nit: can we have a helper function for doing this kind of thing?

(also in general I would prefer the arguments of writeln to not be _super_ complicated, better to hoist this out into a variable first)